### PR TITLE
Scala 3.4.3 support

### PR DESCRIPTION
**Pull request by lm1nrt**
### Description

-  Added artifacts for Scala 3.4.3 version
-  Updated tests that are dependent on the chosen scala version
-  Fixed `java.lang.NoClassDefFoundError: xsbti/UseScope` for Scala versions >= 3.4.0

### Motivation
There are newer versions of Scala that can be supported.

### CODE REVIEW

File: scala/private/macros/scala_repositories.bzl [hunk 1]
```diff
@@ -140,6 +140,7 @@ def _artifact_ids(scala_version):
         "io_bazel_rules_scala_scala_xml",
         "io_bazel_rules_scala_scala_parser_combinators",
         "io_bazel_rules_scala_scala_library_2",
+        "org_scala_sbt_compiler_interface",
```
Comment (lm1nrt) [inline scala/private/macros/scala_repositories.bzl:143]: Added due to `java.lang.NoClassDefFoundError: xsbti/UseScope` for `test_examples.sh`

File: third_party/repositories/scala_3_1.bzl [hunk 2]
```diff
@@ -99,6 +99,10 @@ artifacts = {
             "@io_bazel_rules_scala_scala_library",
         ],
     },
+    "org_scala_sbt_compiler_interface": {
```
Comment (lm1nrt) [inline third_party/repositories/scala_3_1.bzl:102]: What should be better in your opinion - add this artifact for versions that don't need that (3.1.0 =< version < 3.4.0) to stay concise with [artifact_ids](https://github.com/bazelbuild/rules_scala/pull/1603/files#diff-219f92bb35f88485a21a0615e73a715488fa38dbeeda0eb7b9abd251c18573d9L126) or modify `def _artifact_ids(scala_version):`? If modifying, I can do that in next PR.

Comment (simuons) [reply third_party/repositories/scala_3_1.bzl:102]: I'd say keep it as is and do not complicate  `def _artifact_ids(scala_version)` any further. I guess those who use scala 3 will quickly move to never version and older ones will become obsolete.

**Review by simuons — APPROVED** [review]
Thanks @lm1nrt

**Review by liucijus — APPROVED** [review]
Thanks, @lm1nrt!

**Pull request merged**